ambari-server/src/main/java/org/apache/ambari/server/bootstrap/BSHostStatusCollector.java

```diff
@@ -100,7 +100,7 @@ public void run() {
               continue;
             if (line.startsWith("ERROR MESSAGE:") && !status.getStatusCode().equals("0")) {
               // Remove "ERROR MESSAGE:" string
-              status.setError(line.substring(line.indexOf("ERROR MESSAGE:")+(new String("ERROR MESSAGE: ")).length()));
+              status.setError(line.substring((new String("ERROR MESSAGE: ")).length()));
```

Comment:
This change addresses only one of the problems with this part (unnecessary `indexOf()` call).

The remaining problems in decreasing order of importance:

 * unnecessary creation of a `new String` instead of using literal `"ERROR MESSAGE: "`
 * repeating the string `ERROR MESSAGE:` (with the minor variation of an added space) instead of using a constant
 * calculating the length of a fixed string instead of storing it in a constant

Reply:
I use its length directly:

// Remove "ERROR MESSAGE: " string whose length is 15
status.setError(line.substring(15));